src/components/appSwitch/AppSwitch.js

```diff
@@ -37,7 +36,11 @@ class AppSwitch extends React.Component<LoadingProps, {}> {
    */
   constructor(props: LoadingProps) {
     super(props)
-    this.checkAuthStatus()
+    this.ready = false
```

Comment:
I think that would be cleaner if we use 
 `
const setLoading = setLoadingWithStore(this.props.store)
setLoading(true)
` 
Since is the condition that LoadingIndicator is looking for and we will be avoiding maintain double loading states.